arches/app/datatypes/datatypes.py

```diff
@@ -2232,29 +2235,73 @@ def to_json(self, tile, node):
+                    "provisional": provisional,
+                }
+            )
+            if relatedResourceItem.get("resourceName", "") != "":
+                document["strings"].append(
                     {
-                        "id": relatedResourceItem["resourceId"],
+                        "string": relatedResourceItem["resourceName"],
                         "nodegroup_id": tile.nodegroup_id,
                         "provisional": provisional,
                     }
                 )
-                if (
-                    "resourceName" in relatedResourceItem
-                    and relatedResourceItem["resourceName"] not in document["strings"]
-                ):
+            for ontology_property_item in [
+                relatedResourceItem.get("ontologyProperty", ""),
+                relatedResourceItem.get("inverseOntologyProperty", ""),
+            ]:
+                if ontology_property_item != "":
+                    try:
+                        uuid.UUID(ontology_property_item)
+                        relationship = (
+                            self.get_relationship_display_value(ontology_property_item)
+                            or ontology_property_item
+                        )
+                    except ValueError:
+                        relationship = ontology_property_item
                     document["strings"].append(
                         {
-                            "string": relatedResourceItem["resourceName"],
+                            "string": relationship,
                             "nodegroup_id": tile.nodegroup_id,
                             "provisional": provisional,
                         }
                     )
 
+    def get_search_terms(self, nodevalue, nodeid=None):
+        terms = []
+        nodevalue = self.get_nodevalues(nodevalue)
+        lang = get_language()
+        for relatedResourceItem in nodevalue:
+            if relatedResourceItem.get("resourceName", "") != "":
+                terms.append(
+                    SearchTerm(value=relatedResourceItem["resourceName"], lang=lang)
+                )
+            for ontology_property_item in [
+                relatedResourceItem.get("ontologyProperty", ""),
+                relatedResourceItem.get("inverseOntologyProperty", ""),
+            ]:
+                if ontology_property_item != "":
+                    try:
+                        uuid.UUID(ontology_property_item)
+                        relationship = (
+                            self.get_relationship_display_value(ontology_property_item)
+                            or ontology_property_item
+                        )
+                        terms.append(SearchTerm(value=relationship, lang=lang))
+                    except ValueError:
+                        terms.append(
+                            SearchTerm(value=ontology_property_item, lang=lang)
+                        )
```

Comment:
I think we should just leave the lang= kwarg blank "" as we really don't know the language of the values

Reply:
IE 
```
SearchTerm(value=..., lang="")
```